mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala

```diff
@@ -107,9 +107,15 @@ class BisectingKMeansModel private[ml] (
   @Since("2.0.0")
   override def transform(dataset: Dataset[_]): DataFrame = {
     transformSchema(dataset.schema, logging = true)
-    val predictUDF = udf((vector: Vector) => predict(vector))
-    dataset.withColumn($(predictionCol),
-      predictUDF(DatasetUtils.columnToVector(dataset, getFeaturesCol)))
+    if ($(predictionCol).nonEmpty) {
```

Comment:
Likewise how would this be empty, unless the user set it to nothing? in which case, this seems not worth worrying about

Reply:
I changed here to keep in line with other algs like  [LDA.transform](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/clustering/LDA.scala#L461)
Or leave alone algs with only one output column, and remove the check in algs like LDA?

Reply:
Hm, I'd say we don't need this check anywhere that the user would have to explicitly set no prediction column to get no output, and in that case, I don't think it's worth checking and warning. I'm neutral on removing the other checks, but not against it.

Some checks are OK like the ones above as it might be easier to accidentally get into this situation because there are multiple prediction cols.